cli/src/pcluster/config/update_policy.py

```diff
@@ -153,8 +175,60 @@ def __eq__(self, other):
     >= (change.old_value if change.old_value is not None else DEFAULT_MAX_COUNT),
 )
 
+
+def fail_reason_queue_update_strategy(change, _):
+    reason = "All compute nodes must be stopped"
+    # QueueUpdateStrategy can override UpdatePolicy of parameters under SlurmQueues
+    if any(path.startswith("SlurmQueues[") for path in change.path):
+        reason += " or queue update strategy must be set"
+
+    return reason
+
+
+def condition_checker_queue_update_strategy(change, patch):
+    result = not patch.cluster.has_running_capacity()
+    # QueueUpdateStrategy can override UpdatePolicy of parameters under SlurmQueues
+    if any(path.startswith("SlurmQueues[") for path in change.path):
+        result = result or (
+            patch.target_config.get("Scheduling").get("SlurmSettings", {}).get("QueueUpdateStrategy") is not None
```

Comment:
you could simplify the check with:
```python
result = result or patch.target_config.get("Scheduling").get("SlurmSettings", {}).get("QueueUpdateStrategy", None) != QueueUpdateStrategy.COMPUTE_FLEET_STOP.value
```

Reply:
nice, thanks

Reply:
the correct way is to use the default, in case QueueUpdateStrategy is not set
```
result = result or patch.target_config.get("Scheduling").get("SlurmSettings", {}).get("QueueUpdateStrategy", QueueUpdateStrategy.COMPUTE_FLEET_STOP.value) != QueueUpdateStrategy.COMPUTE_FLEET_STOP.value
```